test/e2e/framework/framework.go

```diff
@@ -105,26 +110,76 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
+	// Then hand them off the the std flags and pflags parsers respectively.
+	var testFlags []string
+	for i := 0; i < len(os.Args); i++ {
+		if strings.HasPrefix(os.Args[i], "-test.") {
+			// Add the test flag to our testing flags slice so we can parse it seperatly
+			testFlags = append(testFlags, os.Args[i])
+			// Remove the test flag from our pflags slice
+			os.Args = append(os.Args[:i], os.Args[i+1:]...)
+			// go back one as we just deleted an element
+			i--
+		}
+	}
+
+	// Parse the testing flags
+	return flag.CommandLine.Parse(testFlags)
+}
+
 // NewFromFlags sets up the testing framework with the standard command line flags.
 func NewFromFlags() (*Framework, error) {
-	usr, _ := user.Current()
-	kubeconfig := flag.String("kubeconfig", filepath.Join(usr.HomeDir, "/.kube/config"),
-		"kube config path, e.g. $HOME/.kube/config")
-	gsimage := flag.String("gameserver-image", "gcr.io/agones-images/udp-server:0.19",
-		"gameserver image to use for those tests, gcr.io/agones-images/udp-server:0.19")
-	pullSecret := flag.String("pullsecret", "",
-		"optional secret to be used for pulling the gameserver and/or Agones SDK sidecar images")
-	stressTestLevel := flag.Int("stress", 0, "enable stress test at given level 0-100")
-	perfOutputDir := flag.String("perf-output", "", "write performance statistics to the specified directory")
-	version := flag.String("version", "", "agones controller version to be tested, consists of release version plus a short hash of the latest commit")
-
-	flag.Parse()
+	usr, err := user.Current()
+	if err != nil {
+		return nil, err
+	}
+
+	viper.SetDefault(kubeconfigFlag, filepath.Join(usr.HomeDir, "/.kube/config"))
+	viper.SetDefault(gsimageFlag, "gcr.io/agones-images/udp-server:0.19")
+	viper.SetDefault(pullSecretFlag, "")
+	viper.SetDefault(stressTestLevelFlag, 0)
+	viper.SetDefault(perfOutputDirFlag, "")
+	viper.SetDefault(versionFlag, "")
+	viper.SetDefault(featureGatesFlag, "")
+
+	kubeconfig := pflag.String(kubeconfigFlag, viper.GetString(kubeconfigFlag), "kube config path, e.g. $HOME/.kube/config")
+	gsimage := pflag.String(gsimageFlag, viper.GetString(gsimageFlag), "gameserver image to use for those tests, gcr.io/agones-images/udp-server:0.19")
+	pullSecret := pflag.String(pullSecretFlag, viper.GetString(pullSecretFlag), "optional secret to be used for pulling the gameserver and/or Agones SDK sidecar images")
+	stressTestLevel := pflag.Int(stressTestLevelFlag, viper.GetInt(stressTestLevelFlag), "enable stress test at given level 0-100")
+	perfOutputDir := pflag.String(perfOutputDirFlag, viper.GetString(perfOutputDirFlag), "write performance statistics to the specified directory")
+	version := pflag.String(versionFlag, viper.GetString(versionFlag), "agones controller version to be tested, consists of release version plus a short hash of the latest commit")
+	runtime.FeaturesBindFlags()
+	pflag.Parse()
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```

Comment:
Having env binding is nice, but do we need it for e2e tests?

Reply:
At this moment - no. I will remove it then.